src/index.ts

```diff
@@ -485,7 +485,7 @@ export class AzureOpenAI extends OpenAI {
   }
 
   protected override async prepareOptions(opts: Core.FinalRequestOptions<unknown>): Promise<void> {
-    if (opts.headers?.['Authorization'] || opts.headers?.['api-key']) {
```

Comment:
This is a bit awkward as users could set the `Authorization` header themselves and then we'd override it.

I think a better solution would be to do the same thing as we do in the Python SDK where we create a copy of the options before it's passed to this `prepareOptions()` function

Reply:
The Azure client doesn't support users setting the authorization header manually. The constructor requires either an API key or a bearer token provider function. If the user really wants to set the authorization header manually, they should provide a bearer token provider function that returns just that value.

Reply:
👍 mind adding that context as a comment to this code saying why it's okay to unconditionally overwrite the `Authorization` header?

Reply:
Added in https://github.com/openai/openai-node/pull/1003/commits/e557ea4da4582a8fb9ce20f1d8be481e5a27f08d